gobblin-modules/gobblin-orc/src/test/java/org/apache/gobblin/writer/AvroOrcSchemaConverterTest.java

```diff
@@ -0,0 +1,203 @@
+        .fields()
+        .name("test_union")
+        .type(SchemaBuilder.builder().unionOf().stringType().endUnion())
+        .noDefault()
+        .endRecord();
+    //Ensure that the union type is preserved
+    Assert.assertEquals(AvroOrcSchemaConverter.getOrcSchema(nonNullableAvroUnion), TypeDescription.createStruct()
+        .addField("test_union", TypeDescription.createUnion().addUnionChild(TypeDescription.createString())));
+  }
+
+  @Test
+  public void testTrivialAvroSchemaTranslation() throws Exception {
+
+    // Trivial cases
+    Schema avroSchema = SchemaBuilder.record("test")
+        .fields()
+        .name("string_type")
+        .type(SchemaBuilder.builder().stringType())
+        .noDefault()
+        .name("int_type")
+        .type(SchemaBuilder.builder().intType())
+        .noDefault()
+        .endRecord();
+
+    TypeDescription orcSchema = TypeDescription.createStruct()
+        .addField("string_type", TypeDescription.createString())
+        .addField("int_type", TypeDescription.createInt());
+
+    // Top-level record name will not be replicated in conversion result.
+    Assert.assertEquals(avroSchema.getFields(), getAvroSchema(orcSchema).getFields());
+  }
+
+  @Test
+  public void testUnionAvroSchemaTranslation() throws Exception {
+    Schema avroSchema = SchemaBuilder.record("test")
+        .fields()
+        .name("union_nested")
+        .type(SchemaBuilder.builder().unionOf().stringType().and().intType().endUnion())
+        .noDefault()
+        .endRecord();
+    TypeDescription orcSchema = TypeDescription.createStruct()
+        .addField("union_nested", TypeDescription.createUnion()
+            .addUnionChild(TypeDescription.createString())
+            .addUnionChild(TypeDescription.createInt()));
+
+    Assert.assertEquals(avroSchema.getFields(), getAvroSchema(orcSchema).getFields());
+  }
+
+  @Test
+  public void testSchemaSanitization() throws Exception {
+
+    // Two field along with null
+    Schema avroSchema = SchemaBuilder.builder().unionOf().nullType().and().stringType().and().intType().endUnion();
+    Schema expectedSchema = SchemaBuilder.builder().unionOf().stringType().and().intType().endUnion();
+    Assert.assertEquals(sanitizeNullableSchema(avroSchema), expectedSchema);
+
+    // Only one field except null
+    Schema avroSchema_1 = SchemaBuilder.builder()
+        .unionOf()
+        .nullType()
+        .and()
+        .record("test")
+        .fields()
+        .name("aaa")
+        .type(SchemaBuilder.builder().intType())
+        .noDefault()
+        .endRecord()
+        .endUnion();
+    expectedSchema = SchemaBuilder.builder()
+        .record("test")
+        .fields()
+        .name("aaa")
+        .type(SchemaBuilder.builder().intType())
+        .noDefault()
+        .endRecord();
+    Assert.assertEquals(sanitizeNullableSchema(avroSchema_1), expectedSchema);
+  }
+
+  public static Schema getAvroSchema(TypeDescription schema) {
```

Comment:
A candidate to move into a test utils class?

Reply:
The same argument as the first one ...